Services/OpenIdConnect/classes/class.ilAuthProviderOpenIdConnect.php

```diff
@@ -79,32 +86,30 @@ public function doAuthentication(ilAuthStatus $status) : bool
 
             $oidc->addScope($this->settings->getAllScopes());
             $oidc->addAuthParam(['response_mode' => 'form_post']);
-            switch ($this->settings->getLoginPromptType()) {
-                case ilOpenIdConnectSettings::LOGIN_ENFORCE:
-                    $oidc->addAuthParam(['prompt' => 'login']);
-                    break;
+            if ($this->settings->getLoginPromptType() === ilOpenIdConnectSettings::LOGIN_ENFORCE) {
+                $oidc->addAuthParam(['prompt' => 'login']);
             }
             $oidc->setAllowImplicitFlow(true);
 
             $oidc->authenticate();
             // user is authenticated, otherwise redirected to authorization endpoint or exception
-            $this->getLogger()->dump($_REQUEST, ilLogLevel::DEBUG);
+            $this->logger->dump($this->body, ilLogLevel::DEBUG);
 
             $claims = $oidc->getVerifiedClaims(null);
-            $this->getLogger()->dump($claims, ilLogLevel::DEBUG);
+            $this->logger->dump($claims, ilLogLevel::DEBUG);
             $status = $this->handleUpdate($status, $claims);
 
             // @todo : provide a general solution for all authentication methods
-            $_GET['target'] = (string) $this->getCredentials()->getRedirectionTarget();// TODO PHP8-REVIEW Please eliminate this. Mutating the request is not allowed and will not work in ILIAS 8.
+            $_GET['target'] = $this->getCredentials()->getRedirectionTarget();// TODO PHP8-REVIEW Please eliminate this. Mutating the request is not allowed and will not work in ILIAS 8.
```

Comment:
I am sure I had to also handle this for the `SAML` authentication, maybe you could have a look there.